### DISCUSSION TITLE
vale linter not working with markdown files

### DISCUSSION THREAD

**Discussion by diegoholiveira** [w0rp/ale#4159] [Q&A - Ask for help with problems]
I'm trying to setup `vale` as a linter to markdown file using this configuration:

```vim
let g:ale_linters = {'markdown': ['vale'], 'text': ['vale']}
let g:ale_markdown_vale_options = '--config=docs/vale/.vale.ini'
let g:ale_markdown_vale_input_file = '%s'
```

When I call `:ALEInfo`, `vale` is not being under the enabled linters neither the linter variables is displayed.

```
Current Filetype: markdown
Available Linters: ['alex', 'cspell', 'languagetool', 'markdownlint', 'mdl', 'proselint', 'redpen', 'remark_lint', 'textlint', 'vale', 'writegood']
   Linter Aliases:
'remark_lint' -> ['remark-lint']
'writegood' -> ['write-good']
  Enabled Linters: []
  Ignored Linters: []
 Suggested Fixers:
  'dprint' - Pluggable and configurable code formatting platform
  'pandoc' - Fix markdown files with pandoc.
  'prettier' - Apply prettier to a file.
  'remark-lint' - Fix markdown files with remark-lint
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'textlint' - Fix text files with textlint --fix
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
 Linter Variables:
let g:ale_markdown_remark_lint_executable = 'remark'
let g:ale_markdown_remark_lint_options = ''
let g:ale_markdown_remark_lint_use_global = 0
```

**note**: If I call `echo g:ale_markdown_vale_options`, it display the value defined correctly.

Any ideas about what's happening?

**Comment by hsanson** [comment]
This may be related to this issue:  https://github.com/dense-analysis/ale/issues/3363, check specifically this comment: https://github.com/dense-analysis/ale/issues/3363#issuecomment-1046067503

**Reply by diegoholiveira** [reply]
I'm pretty sure it's not the same issue. Even when I disable my configs to `vale`, `vim-ale` doesn't enable it. And the command history is completely empty (it doesn't check if the executable exists).

**Comment by hsanson** [comment]
> Enabled Linters: []

You ALEInfo shows no linters are enabled.  Make sure to set g:ale_linters variable properly:

```
let g:ale_linters = {
  \   'markdown': ['vale']
\}
```

**Reply by diegoholiveira** [reply]
That's the issue. I have this config in my `vimrc`, yet, vale is not displayed as an enabled linter.
